theano/tensor/nnet/abstract_conv.py

```diff
@@ -1507,6 +1526,13 @@ def conv(self, img, kern, mode="valid", dilation=1):
         out_shape = get_conv_output_shape(img.shape, kern.shape,
                                           mode, [1] * self.convdim, dilation)
 
+        out_rows = out_shape[self.convdim]
+        out_cols = out_shape[self.convdim+1]
```

Comment:
This only works for 2D, so make sure you check that somewhere if you don't want to implement it now (the rest of the function does 3D as well).

I also don't think `self.convdim` is the best thing to use here, it doesn't really make sense. It's just a coincidence that batch and output channels use exactly `self.convdim` dimensions, so I'd just use 2.

Even better: you could make this whole first part work in a convdim-agnostic way, by checking if `kern.shape[2] == np.prod(out_shape[2:])`.